container-host-files/etc/scf/config/scripts/configure-az.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
+
+if test -z "${AZ_LABEL_NAME}"
+then
+    echo AZ: No label configured
+    echo AZ: Skipping
+else
+    # AZ override processing is in effect.
+
+    # Locate the kubectl binary.
+    ##
+    # Attention! This only works because of us having/using a common
+    # layer which contains all the packages, regardless of their use
+    # within a specific role. This gives the diego-cell access to
+    # things it normally is not using at all. Here this is the
+    # `kubectl` cli originally added for use by the
+    # `acceptance-tests-brain`.
```

Comment:
That's… not awesome.

Can we make a dummy job in `scf-release` that pulls in the package, or something along those lines? So that there's an actual dependency to the correct package?

(I was originally thinking pre-start script, but this needs to run before `configgin` does, I think, and that's too late.)

We also make kube calls in [`configure-HA-hosts.sh`](https://github.com/SUSE/scf/blob/develop/container-host-files/etc/scf/config/scripts/configure-HA-hosts.sh), but that uses the HTTP API directly.  `curl` should be in the stemcell (so at least we don't get the magic action-at-a-distance thing).

Also, it looks like we can't use the [downwards API](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/) because that only exposes `nodeName` (not any of the labels on the node).

Reply:
Hm. ... A dummy job might actually work. Thank you. Will have a look.

Reply:
Also, on a general note we should not reference the `packages-src` directory in SCF at all. It is a fissile implementation detail that may already be broken by https://github.com/cloudfoundry-incubator/fissile/pull/439/files

It *moves* the directories from `packages-src` to `packages` to make them available to processes running under `bpm` control.

I'm already working on a PR that removes the 2 references we have to `packages-src` in SCF and UAA, so please don't create any new ones. 😄

Reply:
Right. I wrote the big note as reminder that this would likely break the moment we use separate images per release ... Or something. ... Already broken, good enough to find something less of a hack, or no hack.

Reply:
The `cfdot` job in `diego-release` is an example of a job used to just install a binary (and update a few config files): https://github.com/cloudfoundry/diego-release/tree/develop/jobs/cfdot

Reply:
ty for the reference.

Reply:
Done with commit 8b7361f102776fb116cbce38b94475d58df2b6f8